base/sparse/sparsevector.jl

```diff
@@ -1492,3 +1492,15 @@ function _At_or_Ac_mul_B{TvA,TiA,TvX,TiX}(tfun::BinaryOp, A::SparseMatrixCSC{TvA
     end
     SparseVector(n, ynzind, ynzval)
 end
+
+#sorting
+function sort{Ti,Tv}(x::SparseVector{Ti,Tv}; kws...)
```

Comment:
`Ti` and `Tv` should be switched here; your tests passed because they're both `Int`s. I think it would be worth using a `Float64` sparse vector in your tests to catch stuff like this.